docs/glossary.md

```diff
@@ -31,3 +31,19 @@
   - Defending units can only participate in this phase if `Submarines Defending May Submerge Or Retreat` property is set.
   - `Sub Retreat Before Battle` defines if this phase occurs before the First Strike Phase or at the end of the Battle Round.
 - **Submerge Subs Phase**: Another name for **Evasion Phase** and generally used when only submarines have `canEvade` and `Submersible Subs` is set.
+
+##Development Terms
+- **Branch**: A Branch is a copy of the main codebase.  It is used to keep changes separate until they are completed or in a stable state.  
+- **Checkout**: When you Checkout a branch, that is where commits are stored.  
+- **Clone**: Cloning is when you take your repository and copy it to your local machine to work on.  
+- **Commit**: A Commit is a list of changes being made to a repository.  
+- **Fetch**: When you run a fetch command, your IDE is updating information in your local project.  
+- **Fork**: A Fork is when you make your own copy of a repository in Version Control (Github for TripleA). 
+- **IDE**: And IDE is an Integrated Development Environment that is a software application that combines many of the tools needed by programmers.  Examples are IntelliJ and Eclipse.  
+- **Origin**: The Origin is your fork of another repository.  
+- **Pull Request**: A Pull Request is a request to incorporate a set of commits into the main repository.  
+- **Push**: A Push is when you move a set of commits from your local work space to your repository.  
+- **Repository**:  A Repository is the place where TripleA game and map files are stored.  
```

Comment:
A repository is a set of commits and a set of branch pointers, and the binary data of the file contents stored in a 'git' format. Git stores the data of files as a series of differences, from one commit to another. These 'deltas' are highly optimized for compact storage. For example, the deltas stored are not necessarily the deltas between commits in a branch. For example, one branch may have commits "{A , B, C}", and another "{X, Y, Z}", git might store the delta from B to C as being the sum of deltas from {B, Y, C}. 'Git compaction' is the  operation to optimize this graph of deltas and 
prune is an operation to drop orphaned deltas. Hence, just because you delete a branch, it does not mean all of the commits from that branch would disappear.